### PR TITLE
Allow model parameters to be set through the message interface

### PR DESCRIPTION
**Pull request by rplzzz**
Currently supported parameters are:

* Climate sensitivity
* Beta
* Q10
* preindustrial CO2 concentration
* Diffusivity

### CODE REVIEW

**Comment by rplzzz** [comment]
Looks like I forgot to test changing the diffusivity.  I'll add that in a jiffy.

File: inst/include/carbon-cycle-model.hpp [hunk 1]
```diff
@@ -25,8 +25,8 @@
 // 1 ppm by volume of atmosphere CO2 = 2.13 Gt C, from http://cdiac.ornl.gov/pns/convert.html
 //  and Wigley (1993)
 // So to convert 1 Pg C to 1 ppmv CO2...
-#define PGC_TO_PPMVCO2   (1/2.13)
-
+#define PGC_TO_PPMVCO2 (1.0/2.13)
+#define PPMVCO2_TO_PGC (2.13/1.0)
```
Comment (bpbond) [inline inst/include/carbon-cycle-model.hpp]: maybe `1.0 / PGC_TO_PPMVCO2`

File: inst/include/simpleNbox.hpp [hunk 2]
```diff
@@ -197,6 +197,7 @@ class SimpleNbox : public CarbonCycleModel {
     unitval sum_map( unitval_stringmap pool ) const;    //!< sums a unitval map (collection of data)
     double sum_map( double_stringmap pool ) const;      //!< sums a double map (collection of data)
     void log_pools( const double t );                   //!< prints pool status to the log file
+    void set_c0(double newc0);                          //!< set initial co2 and adjust total carbon mass
```
Comment (bpbond) [inline inst/include/simpleNbox.hpp]: 👍

File: src/ocean_csys.cpp [hunk 3]
```diff
@@ -50,7 +50,7 @@ oceancsys::oceancsys() : ncoeffs(6), m_a(ncoeffs) {
 }
 
 //------------------------------------------------------------------------------
-/*! \brief A helper functor class used to evauluate a polynomial and it's derivative.
+/*! \brief A helper functor class used to evauluate a polynomial and its derivative.
```
Comment (bpbond) [inline src/ocean_csys.cpp]: 🎉

File: src/simpleNbox.cpp [hunk 4]
```diff
@@ -334,20 +343,20 @@ void SimpleNbox::prepareToRun() throw( h_exception )
...
         unitval alb( -0.2, U_W_M2 ); // default is MAGICC value
         Ftalbedo.set( core->getStartDate(), alb );
         Ftalbedo.set( core->getEndDate(), alb );
     }
-    
-    Ca.set( C0.value( U_PPMV_CO2 ), U_PPMV_CO2 );
+
+    double c0init = C0.value(U_PPMV_CO2);
+    Ca.set(c0init, U_PPMV_CO2);
+    atmos_c.set(c0init * (1.0/PGC_TO_PPMVCO2), U_PGC);
```
Comment (bpbond) [inline src/simpleNbox.cpp]: If defining `PGC_TO_PPMVCO2` use it

Comment (rplzzz) [reply src/simpleNbox.cpp]: Done

**Pull request merged**